src/Homework1.java

```diff
@@ -1,101 +1,222 @@
+    static Map<String, Float> getMaxMinAndAverage(int[] array) {
+        Map<String, Float> map = new HashMap<String, Float>();
+        if (array.length >= 1) {
+            int sum = array[0];
+            int max = array[0];
+            int min = array[0];
+            for (int i = 1; i < array.length; i++) {
+                sum += array[i];
+                if (array[i] < min)
+                    min = array[i];
+                else if (array[i] > max)
+                    max = array[i];
+            }
+            float average = (float) sum / array.length;
+            map.put("Max", (float) max);
+            map.put("Min", (float) min);
+            map.put("Average", average);
+        }
+        return map;
+    }
+
+    static int getIndex(int[] array, int target) {
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == target)
+                return i;
+        }
+        return -1;
+    }
+
+    static void findNumberOfOddAndEvenNumbers(int[] array) {
+        int oddNumbers = 0, evenNumbers = 0;
+        for (int i = 0; i < array.length; i++) {
+            if ((array[i] & 1) == 0)
+                evenNumbers++;
+            else
+                oddNumbers++;
+        }
+        System.out.println("EvenNumbers: " + evenNumbers);
+        System.out.println("OddNumbers: " + oddNumbers);
+    }
+
+    // Rabin-Karp Substring Search
+    static boolean isString2InString1(String s1, String s2) {
+        if (s2.length() > s1.length())
+            return false;
+
+        int s2HashCode = s2.hashCode();
+        int substringHashCode = s1.substring(0, s2.length()).hashCode();
+        if (substringHashCode == s2HashCode && s1.substring(0, s2.length()).equals(s2)) {
+            return true;
         }
 
-//        if (myVar % 2 != 0) {
-//            System.out.println("The variable is odd!");
-//        }
-
-        // Display week-days
-        byte weekDay = 50;
-
-        switch (weekDay) {
-            case 1:
-                System.out.println("Mon");
-                break;
-            case 2:
-                System.out.println("Tue");
-                break;
-            case 3:
-                System.out.println("Wen");
-                break;
-            case 4:
-                System.out.println("Thu");
-                break;
-            case 5:
-                System.out.println("Fri");
-                break;
-            case 6:
-                System.out.println("Sat");
-                break;
-            case 7:
-                System.out.println("Sun");
-                break;
-            default:
-                System.out.println("Wrong number: " + weekDay);
+        for (int i = 1; i <= s1.length() - s2.length(); i++) {
+            substringHashCode = substringHashCode - s1.charAt(i - 1) * (int) Math.pow(31, s2.length() - 1);
+            substringHashCode = substringHashCode * 31 + s1.charAt(i - 1 + s2.length());
```

Comment:
Ce se intampla daca algoritmul meu de hashCode foloseste alt numar prim pentru generearea hash-ului in loc de 31?

Reply:
As avea hashcode(string1) != hascode(string 2), desi string1.equals(string2).
Multumesc pentru atentionare. Are sens ca pe viitor functia hashcode sa poate fi modificata, fapt ce-ar duce la o eroare.